src/service/product.service.ts

```diff
@@ -65,6 +65,33 @@ export class ProductService {
     });
 
     return updatedItem;
+  };
+  public static async getProductsByVendorId(vendor_id: string): Promise<Product[]> {
+    try {
+      const products = await Product.findAll({
+        where:{vendor_id},
+        include: [
+          Category,
+          Vendor
+        ]
+      })
+      return products;
+    } catch (error) {
+      throw new Error('Internal Server Error');
+    }
+  }
+  public static async UsergetAllProducts(): Promise<Product[]> {
+    try {
+      const products = await Product.findAll({
+        include: [
+          Category,
+          Vendor
+        ]
+      })
+      return products;
+    } catch (error) {
+      throw new Error('Internal Server Error');
+    }
```

Comment:
Here you're just fetching all products right? Yet in the instructions you were told that you should get all products that are available if the requesting user is a buyer and you should a seller's collection if the request is from a seller.

Reply:
yes but , i have to services, one is that one that fetch all products when a user is a buyer, and i have another one to fetch product by vendor_id and product_id, when the request is from the seller

Reply:
![Screenshot from 2024-05-20 16-56-19](https://github.com/atlp-rwanda/e-commerce-serge-30-bn/assets/126393904/3efe6628-25ce-4757-951f-68b0c17fd524)